tutorials/plot_object_annotations.py

```diff
@@ -0,0 +1,172 @@
+marks bad segments due to eye blinks.
+"""
+
+import os.path as op
+import numpy as np
+from datetime import datetime
+
+import mne
+
+# Load the data
+data_path = mne.datasets.sample.data_path()
+fname = op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw.fif')
+raw = mne.io.read_raw_fif(fname)
+
+# Plot the events
+events = mne.find_events(raw)
+
+# Specify event_id dictionary based on the experiment
+event_id = {'Auditory/Left': 1, 'Auditory/Right': 2,
+            'Visual/Left': 3, 'Visual/Right': 4,
+            'smiley': 5, 'button': 32}
+color = {1: 'green', 2: 'yellow', 3: 'red', 4: 'c', 5: 'black', 32: 'blue'}
+
+mne.viz.plot_events(events, raw.info['sfreq'], raw.first_samp, color=color,
+                    event_id=event_id)
+
+# Create some annotations
+annotated_blink_raw = raw.copy()
+eog_events = mne.preprocessing.find_eog_events(raw)
+n_blinks = len(eog_events)
+# Center to cover the whole blink with full duration of 0.5s:
+onset = eog_events[:, 0] / raw.info['sfreq'] - 0.25
+duration = np.repeat(0.5, n_blinks)
+annot = mne.Annotations(onset, duration, ['bad blink'] * n_blinks,
+                        orig_time=raw.info['meas_date'])
+annotated_blink_raw.set_annotations(annot)
+
+annotated_blink_raw.plot()  # plot the annotated raw
+
+
+###############################################################################
+# Working with Annotations
+# ------------------------
+#
+# An important element of :class:`Annotations <mne.Annotations>` is
+# ``orig_time`` which is the time reference for the ``onset``.
+# It is key to understand that when calling
+# :func:`raw.set_annotations <mne.io.Raw.set_annotations>`, the given
+# annotations are copied and transformed so that
+# :class:`raw.annotations.orig_time <mne.Annotations>`
+# matches the recording time of the raw object.
+# Refer to the documentation of :class:`Annotations <mne.Annotations>` to see
+# the expected behavior depending on ``meas_date`` and ``orig_time``.
+# Where ``meas_date`` is the recording time of the stored in
+# :class:`Info <mne.Info>`.
+# You can find more information about :class:`Info <mne.Info>` in
+# :ref:`sphx_glr_auto_tutorials_plot_info.py`.
+#
+# We'll now manipulate some simulated annotations.
+# The first annotations has ``orig_time`` set to ``None`` while the
+# second is set to a chosen POSIX timestamp for illustration purposes.
+
+###############################################################################
+
+# Create an annotation object without orig_time
+annot_none = mne.Annotations(onset=[0, 2, 9], duration=[0.5, 4, 0],
+                             description=['foo', 'bar', 'foo'],
+                             orig_time=None)
+print(annot_none)
+
+# Create an annotation object with orig_time
+orig_time = '2002-12-03 19:01:31.676071'
+my_datetime = datetime.strptime(orig_time, '%Y-%m-%d %H:%M:%S.%f')
+posix_timestamp = (my_datetime - datetime(1970, 1, 1)).total_seconds()
```

Comment:
this is not used in the annotations object

Reply:
No I used only for printing. There's a subtiliy. the `orig_time` has to be `.6760709` otherwise they wont match.

Reply:
well okay but you don't use this `orig_time` either ...

Reply:
```py
print(posix_timestamp - annot_orig.orig_time)
```
```sh
1.1920928955078125e-07
```

I am using the same number. (except the rounding error)

Reply:
`1038942091.676071 is the POSIX timestamp of 2002-12-03 19:01:31.676071`

it only differs in the `.6760709`

Reply:
okay maybe then it's a matter of renaming your variables. Because I see this:

```py
orig_time = '2002-12-03 19:01:31.676071'
```

and I'm confused why in the class instantiation you do: `orig_time=1038942091.6760709`.

also how much does the 1e-7 matter?

Reply:
@jasmainak I agree it's probably too much to ask to our user to understand this gymnastic of dates.
I am going to push now a way to pass directly '2002-12-03 19:01:31.676071' to the Annotations constructor.

Reply:
actually I need #5699 first to do this... let me switch to #5699 then.